format specifies type 'unsigned short' but the argument has type 'int' error in network stack

**Describe the bug**
Build failure when building with llvm:

```
[ 82%] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/connection.c.obj
[ 83%] Building C object zephyr/subsys/net/ip/CMakeFiles/subsys__net__ip.dir/tcp2.c.obj
/home/nashif/Work/zephyrproject/zephyr/subsys/net/ip/tcp2.c:885:2: error: format specifies type 'unsigned short' but the argument has type 'int' [-Werror,-Wformat]
        conn_send_data_dump(conn);
        ^~~~~~~~~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/subsys/net/ip/./tcp2_priv.h:102:3: note: expanded from macro 'conn_send_data_dump'
                conn_mss((_conn)));                                     \
                ^~~~~~~~~~~~~~~~~~
/home/nashif/Work/zephyrproject/zephyr/include/net/net_core.h:49:6: note: expanded from macro 'NET_DBG'
                        ##__VA_ARGS__)
```

**To Reproduce**

export ZEPHYR_TOOLCHAIN_VARIANT=llvm
west build -b qemu_x86 tests/net/socket/poll

**Expected behavior**
clean build

--- a/subsys/net/ip/tcp2_priv.h
+++ b/subsys/net/ip/tcp2_priv.h
@@ -83,7 +83,7 @@
 
 #define conn_mss(_conn)					\
 	((_conn)->recv_options.mss_found ?		\
-	 (_conn)->recv_options.mss : NET_IPV6_MTU)
+	 (_conn)->recv_options.mss : (uint16_t)NET_IPV6_MTU)
 
 #define conn_state(_conn, _s)						\
 ({									\

